Jesse camden add search nutrition filtering

camdenf101 opened the pull request:
Ensure all acceptance criteria are met.

jjjjess committed:
Feat Added: Inputs for nutritional filtering

- added text inputs for nutritional filtering. Includes: Calories, Carbs, Sodium, and Sugar
- added CSS to get inputs center aligned and spaced out over page

jjjjess committed:
Feat: more functionality added to nutritionFiltering

- Created React Hook for nutritionFilters, setNutritionFilters
- Touched up Nutrition Filters label/input design, added different identifiers, like type, name,id, value, min, and onChange
- Added comments for where I'm stuck, and what I assume needs to be done

camdenf101 committed:
Merge branch 'main' into jesse-camden-add-search-nutrition-filtering

camdenf101 committed:
feat: add nutrition filter

* Changes were made to enable searching by nutrition filters.

camdenf101 committed:
feat: add nutrition parameters to search_recipes

*Changes were made to add functionality to search by nutrition filters.
* Added parameters for max calories, carbs, sodium, and sugar

camdenf101 committed:
feat: update getRecipes function for nutrition.

* Changes were made to update the getRecipes function.
*Add nutrition filters parameters
*Change the approach the URL is built.
*add non-empty params to params array

camdenf101 committed:
fix: update search_recipes.php to send an error.

* Changes made to send an error to frontend if no recipes with filters
*Add nutrition flag

camdenf101 committed:
fix: update getRecipes function to take in params.

*Changes made to update the function for error handling and params array.
*Parameters are passed to function as array
*Add nutrition data to response if necessary

camdenf101 committed:
fix: add error message if no recipes with filters

*Changes made to add a alert if no recipes found with selected filters.

amoots-emich reviewed (changes requested):
This breaks the ability to search without being logged in. Changes required:

Searching without a user token (as long as pantry filtering isn't requested)
Returning a JSON error message if pantry filtering is requested without a token or pantry items. Frontend expects the following message: "Empty User Pantry"

camdenf101 committed:
fix: update search_recipes to be able to merge

* Changes made to fix the ability to search without being logged in.
*Only checks for token and ingredients if pantry filtering is requested

camdenf101 committed:
fix: update getRecipes function to remove userID

*Changes made to fix the ability to search without being logged in.
*Remove UserID parameter.

amoots-emich reviewed (approved):
Tested and working on production.
